route/build/src/process_ntopo.f90

```diff
@@ -419,6 +419,19 @@ subroutine put_data_struct(nSeg, structSEG, structNTOPO, &
      RPARAM_in(iSeg)%D03_MaxStorage  =     structSEG(iSeg)%var(ixSEG%D03_MaxStorage)%dat(1)
      RPARAM_in(iSeg)%D03_Coefficient =     structSEG(iSeg)%var(ixSEG%D03_Coefficient)%dat(1)
      RPARAM_in(iSeg)%D03_Power       =     structSEG(iSeg)%var(ixSEG%D03_Power)%dat(1)
+
+     RPARAM_in(iSeg)%HYP_E_emr       =     structSEG(iSeg)%var(ixSEG%HYP_E_emr)%dat(1)
+     RPARAM_in(iSeg)%HYP_E_lim       =     structSEG(iSeg)%var(ixSEG%HYP_E_lim)%dat(1)
+     RPARAM_in(iSeg)%HYP_E_min       =     structSEG(iSeg)%var(ixSEG%HYP_E_min)%dat(1)
+     RPARAM_in(iSeg)%HYP_E_zero      =     structSEG(iSeg)%var(ixSEG%HYP_E_zero)%dat(1)
+     RPARAM_in(iSeg)%HYP_Qrate_emr   =     structSEG(iSeg)%var(ixSEG%HYP_Qrate_emr)%dat(1)
+     RPARAM_in(iSeg)%HYP_Erate_emr   =     structSEG(iSeg)%var(ixSEG%HYP_Erate_emr)%dat(1)
+     RPARAM_in(iSeg)%HYP_Qrate_prim  =     structSEG(iSeg)%var(ixSEG%HYP_Qrate_prim)%dat(1)
+     RPARAM_in(iSeg)%HYP_Qrate_amp   =     structSEG(iSeg)%var(ixSEG%HYP_Qrate_amp)%dat(1)
+     RPARAM_in(iSeg)%HYP_Qrate_phs   =     structSEG(iSeg)%var(ixSEG%HYP_Qrate_phs)%dat(1)
+     RPARAM_in(iSeg)%HYP_prim_F      =     structSEG(iSeg)%var(ixSEG%HYP_prim_F)%dat(1)
```

Comment:
RPARAM_in(iSeg)%HYP_prim_F is logical array.  gnu compiler complains. structSEG(iSeg)%var(ixSEG%HYP_prim_F)%dat(1) -> (structSEG(iSeg)%var(ixSEG%HYP_prim_F)%dat(1)==1).

Reply:
should have been resolved. please check